src/auditing.coffee

```diff
@@ -70,3 +77,48 @@ exports.processAudit = (msg, callback) ->
       if xmlErr then logger.info "Failed to parse message as an AuditMessage XML document: #{xmlErr}"
 
       callback()
+
+
+sendUDPAudit = (msg, callback) ->
+  client = dgram.createSocket('udp4')
+  client.send msg, 0, msg.length, config.auditing.auditEvents.port, config.auditing.auditEvents.host, (err) ->
+    client.close()
+    callback err
+
+sendTLSAudit = (msg, callback) ->
+  tlsAuthentication.getServerOptions true, (err, options) ->
+    return callback err if err
+
+    client = tls.connect config.auditing.auditEvents.port, config.auditing.auditEvents.host, options, ->
+      if not client.authorized then return callback client.authorizationError
+
+      client.write "#{msg.length} #{msg}"
+      client.end()
+
+    client.on 'error', (err) -> logger.error err
+    client.on 'close', -> callback()
+
+sendTCPAudit = (msg, callback) ->
+  client = net.connect config.auditing.auditEvents.port, config.auditing.auditEvents.host, ->
+    client.write "#{msg.length} #{msg}"
+    client.end()
+
+  client.on 'error', (err) -> logger.error
+  client.on 'close', -> callback()
+
+
+# Send an audit event
+exports.sendAuditEvent = (msg, callback=(->)) ->
+  done = (err) ->
+    if err then logger.error err
+    callback()
+
+  if not config.auditing?.auditEvents?
+    return done 'Unable to record audit event: Missing config.auditing.auditEvents'
+
+  switch config.auditing.auditEvents.interface
+    when 'internal' then processAudit msg, done
+    when 'udp' then sendUDPAudit msg, done
+    when 'tls' then sendTLSAudit msg, done
+    when 'tcp' then sendTCPAudit msg, done
+    else done "Invalid audit event interface '#{config.auditing.auditEvents.interface}'"
```

Comment:
I think its's best to create an [Error object](https://nodejs.org/api/errors.html) here instead of a string.